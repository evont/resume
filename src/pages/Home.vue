<template>
  <div class="home">
    <div class="home-logo" :data-status="isMounted ? 1 : 0"></div>
    <div class="home-option" :data-status="isMounted ? 1 : 0">
      <router-link class="home-option-link" to="/Animate">查看动态简历</router-link>
      <router-link class="home-option-link" to="/Static">查看静态简历</router-link>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Home',
  data() {
    return {
      isMounted: false,
    };
  },
  mounted() {
    this.isMounted = true;
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="postcss" scoped>
  .home {
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    margin: auto;
    height: 200px;
    max-width: 480px;/* no */
    text-align: center;
    &-logo {
      height: 100px;
      width: 100px;
      margin: 20px auto;
      background: url('/static/img/logo.png') no-repeat center;
      background-size: contain;
      opacity: 0;
      transform: translate(0, 20px);
      transition: all .6s;
      &[data-status='1'] {
        opacity: 1;
        transform: translate(0, 0);
      }
    }
    &-option {
      padding: 20px 0;
      opacity: 0;
      transform: translate(0, -20px);
      transition: all .6s .6s;
      &[data-status='1'] {
        opacity: 1;
        transform: translate(0, 0);
      }

      &-link {
        padding: 0 10px;
        color: #369;
      }
    }
  }
</style>
