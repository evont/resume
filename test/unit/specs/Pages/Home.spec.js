import { mount } from 'vue-test-utils'
import Home from '@/pages/Home';

describe('Home.vue', () => {

  it('should be mounted', () => {
    let cmp = mount(Home, {
      data: {
        isMounted: true
      }
    })
    expect(cmp.vm.isMounted).toEqual(true);
  });

  it('should be status euals one', () => {
    let cmp = mount(Home, {
      data: {
        isMounted: true
      }
    })
    expect(cmp.find('.home-logo').attributes()['data-status']).toBe('1')
    expect(cmp.find('.home-option').attributes()['data-status']).toBe('1')
  });

  it('Should have the expected html structure', () => {
    let cmp = mount(Home, {
      data: {
        isMounted: true
      }
    })
    expect(cmp.element).toMatchSnapshot() // 调用toMatchSnapshot来比对快照
  })
});
  