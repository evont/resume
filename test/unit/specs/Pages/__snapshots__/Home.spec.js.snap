// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue Should have the expected html structure 1`] = `
<div
  class="home"
>
  <div
    class="home-logo"
    data-status="1"
  />
   
  <div
    class="home-option"
    data-status="1"
  >
    <router-link
      class="home-option-link"
      to="/Animate"
    >
      查看动态简历
    </router-link>
     
    <router-link
      class="home-option-link"
      to="/Static"
    >
      查看静态简历
    </router-link>
  </div>
</div>
`;
